s3_artifact/action.py

```diff
@@ -0,0 +1,101 @@
+import os
+import subprocess
+from dataclasses import dataclass
+from distutils.util import strtobool
+from pathlib import Path
+from typing import Optional, Sequence
+
+from dataclasses_json import dataclass_json
+from yaml import safe_load
+
+DEFAULT_MIME_TYPES = {
+    ".css": "text/css",
+    ".html": "text/html",
+    ".js": "application/javascript",
+    ".jpg": "image/jpeg",
+    ".jpeg": "image/jpeg",
+    ".pdf": "application/pdf",
+    ".png": "image/png",
+}
+
+S3_NO_PROGRESS = "--no-progress"
+S3_CP_OPTIONS = f"--recursive {S3_NO_PROGRESS}"
+S3_SYNC_OPTIONS = f"--delete {S3_NO_PROGRESS}"
+
+
+@dataclass_json
+@dataclass
+class WebsiteCustomMetadataConfig:
```

Comment:
The name of this and the `WebsiteConfig` class should probably be something like
```suggestion
class S3ArtifactCustomMetadataConfig:
```